client/asset/eth/eth.go

```diff
@@ -5104,25 +5108,20 @@ func transactionFeeLimit(tx *types.Transaction) *big.Int {
 
 // extendedTx generates an *extendedWalletTx for a newly-broadcasted tx and
 // stores it in the DB.
-func (w *assetWallet) extendedTx(tx *types.Transaction, txType asset.TransactionType, amt uint64) *extendedWalletTx {
+func (w *assetWallet) extendedTx(tx *types.Transaction, txType asset.TransactionType, amt uint64, recipient *string) *extendedWalletTx {
 	var tokenAssetID *uint32
 	if w.assetID != w.baseChainID {
 		tokenAssetID = &w.assetID
 	}
-	return w.baseWallet.extendAndStoreTx(tx, txType, amt, tokenAssetID)
+	return w.baseWallet.extendAndStoreTx(tx, txType, amt, tokenAssetID, recipient)
 }
 
-func (w *baseWallet) extendAndStoreTx(tx *types.Transaction, txType asset.TransactionType, amt uint64, tokenAssetID *uint32) *extendedWalletTx {
+func (w *baseWallet) extendAndStoreTx(tx *types.Transaction, txType asset.TransactionType, amt uint64, tokenAssetID *uint32, recipient *string) *extendedWalletTx {
 	nonce := tx.Nonce()
 	rawTx, err := tx.MarshalBinary()
 	if err != nil {
 		w.log.Errorf("Error marshaling tx %q: %v", tx.Hash(), err)
 	}
-	var recipient *string
-	if to := tx.To(); to != nil {
-		s := to.String()
-		recipient = &s
-	}
```

Comment:
Is it worth doing something like ?
```Go
if recipient == nil {
	if to := tx.To(); to != nil {
		s := to.String()
		recipient = &s
	}
}
```

Reply:
You don't think it'll be confusing to have the swap contract address / bond contract address there?

Reply:
For swaps, we should be passing the recipient, no?

Reply:
Do you mean the counterparty's address?

Reply:
I think I'm not fully grasping the intent here. Where's the "bug" exactly?

Reply:
The recipient field was supposed to show the recipient of a "send". For tokens, this will show the token contract address instead of the address where the tokens were sent.

Reply:
I guess I don't see a problem with having the contract as the recipient field, especially if we're just leaving it blank otherwise. More data is better.

Reply:
OK I guess it can't hurt.